src/frequenz/sdk/actor/_decorator.py

```diff
@@ -198,19 +198,23 @@ async def _start_actor(self) -> None:
 
             Raises:
                 asyncio.CancelledError: when the actor's task gets cancelled.
+
+            # noqa: DAR401 cancelled_error
             """
             logger.debug("Starting actor: %s", cls.__name__)
             number_of_restarts = 0
+            cancelled_error = None
             while True:
                 try:
                     await super().run()
-                except asyncio.CancelledError:
+                except asyncio.CancelledError as ex:
+                    cancelled_error = ex
                     logger.debug("Cancelling actor: %s", cls.__name__)
-                    raise
-                except Exception as err:  # pylint: disable=broad-except
-                    logger.exception(
-                        "Actor (%s) crashed with error: %s", cls.__name__, err
-                    )
+                except Exception:  # pylint: disable=broad-except
+                    logger.exception("Actor (%s) crashed", cls.__name__)
+                finally:
+                    if cancelled_error:
+                        raise cancelled_error
```

Comment:
Yes, that's right. The exception should be raised in the except clause. However I'd need a flag to check for CancelledError in the finally clause as the break disables the exception given the current logic in the exception handling. Actually I'm leaning towards to change the current retry-mechanism so that there is not logic needed when handling the exception